dev-clean/headers/llsc/external.hpp

```diff
@@ -0,0 +1,16 @@
+/* LLSC - External utility functions and library modeling functions */
+/* Generated by sai.llsc.TestGenerateExternal */
+#ifndef LLSC_EXTERNAL_HEADERS_GEN
+#define LLSC_EXTERNAL_HEADERS_GEN
+
+/************* Function Declarations **************/
+immer::flex_vector<std::pair<SS, PtrVal>> open(SS, immer::flex_vector<PtrVal>);
+
+/************* Functions **************/
+inline immer::flex_vector<std::pair<SS, PtrVal>> open(SS x1, immer::flex_vector<PtrVal> x2) {
+PtrVal x3 = x2.at(0);
+FS x4 = x1.get_fs();
+x1.set_fs(x4);
+return immer::flex_vector<std::pair<SS, PtrVal>>{std::make_pair(x1, make_IntV(x4.open_file(get_string(x3, x1), 0), 32))};
```

Comment:
I believe there is some ordering violation? i.e., `set_fs` should happen after some status change like `open_file`